src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -78,13 +94,35 @@ constructor a_fresh statements
    = Nothing
 
   pair x
+   | XValue _ (PairT ta tb) (VPair a b) <- x
+   = Just (ta, tb, xValue ta a, xValue tb b)
+
    | Just (PrimMinimal (Min.PrimConst (Min.PrimConstPair ta tb)), [a,b]) <- takePrimApps x
-   = Just (ta,tb,a,b)
+   = Just (ta, tb, a, b)
+
+   | otherwise
+   = Nothing
+
+  option x
+   | XValue _ (OptionT tv) VNone <- x
+   = Just (tv, xFalse, xDefault tv)
+
+   | Just (PrimOption (PrimOptionPack tv), [b, v]) <- takePrimApps x
+   = Just (tv, b, v)
+
+   | Just (PrimMinimal (Min.PrimConst (Min.PrimConstSome tv)), [v]) <- takePrimApps x
+   = Just (tv, xTrue, v)
+
    | otherwise
    = Nothing
 
+
   get env n
    | (_,x'):_ <- filter ((==n).fst) env
    = Just x'
    | otherwise
    = Nothing
+
+  -- ANF?
+  get' env (XVar _ n) = get env n
+  get' _    xx        = Just xx
```

Comment:
Yeah I believe so, because [simp](https://github.com/ambiata/icicle/blob/master/src/Icicle/Avalanche/Simp.hs#L33) is run over the expressions

I'll try doing it again after melt and see if I can remove `get'`